scripts/generate_makefile_deps.R

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env RScript
+
+# generate the dependencies of the PEcAn packages on each other.
+
+files <-
+  c(
+    list.files(
+      path = "base",
+      full.names = TRUE,
+      pattern = "^DESCRIPTION$",
+      recursive = TRUE
+    ),
+    list.files(
+      path = "modules",
+      full.names = TRUE,
+      pattern = "^DESCRIPTION$",
+      recursive = TRUE
+    ),
+    list.files(
+      path = "models",
+      full.names = TRUE,
+      pattern = "^DESCRIPTION$",
+      recursive = TRUE
+    )
+  )
+
+## Required dependencies
+pecan <- c()
+depends <- c()
+d <- purrr::walk(files,
+                 function(x) {
+                   y <- desc::desc_get_deps(x)
+                   y <- y[y$type %in% c('Depends', 'Imports', 'Remotes'), 'package']
+                   y <- y[grepl('^PEcAn.', y)]
+                   p <- desc::desc_get_field('Package', file=x)
+                   f <- gsub('/DESCRIPTION$', '', x)
+                   pecan[[p]] <<- f
+                   depends[[f]] <<- y
+                 } )
+
+for (name in names(depends)) {
+  x <- paste0('$(call depends,', name, '): |')
+  for (p in depends[[name]]) {
+    x <- paste0(x, ' .install/', pecan[[p]])
+  }
+  print(x)
```

Comment:
`print` adds R's vector notation and quotes (`# [1] "your string"`). Message prints directly to `stdout`. You could also do `cat(x, "\n")`.

```suggestion
  message(x)
```

Reply:
now writing to Makefile.depends which included in the Makefile